src/javascript/app/pages/user/new_account/welcome_page.js

```diff
@@ -1,50 +1,99 @@
 const Client           = require('../../../base/client');
 const getElementById   = require('../../../../_common/common_functions').getElementById;
-const localize         = require('../../../../_common/localize').localize;
-const Url              = require('../../../../_common/url');
-const createElement    = require('../../../../_common/utility').createElement;
 const showLoadingImage = require('../../../../_common/utility').showLoadingImage;
+const ClientBase       = require('../../../../_common/base/client_base');
+const urlFor           = require('../../../../_common/url').urlFor;
 
 const WelcomePage = (() => {
+
+    let el_welcome_container,
+        is_virtual,
+        upgrade_info,
+        is_uk,
+        is_unwelcome_uk,
+        cfd,
+        d_options,
+        not_sure;
+
+    const init = () => {
+        upgrade_info          = ClientBase.getBasicUpgradeInfo();
+        is_virtual            = Client.get('is_virtual');
+        el_welcome_container  = getElementById('welcome_container');
+        is_uk                 = Client.get('residence') === 'gb';
+        is_unwelcome_uk       = Client.isUnwelcomeUk();
+        not_sure              = getElementById('default');
+        cfd                   = getElementById('cfd');
+        d_options             = getElementById('d_ptions');
+    };
+    
+    const getCanUpgrade = (upgrade_type,  { can_upgrade_to } = upgrade_info) => can_upgrade_to.includes(upgrade_type);
+  
     const onLoad = () => {
-        BinarySocket.wait('authorize', 'landing_company', 'get_settings').then(() => {
-            const el_welcome_container = getElementById('welcome_container');
+        BinarySocket.wait('authorize', 'landing_company', 'get_settings', 'get_account_status').then(() => {
+            init();
+
             if (Client.hasAccountType('real')) {
-                window.location.href = Client.defaultRedirectUrl();
+                BinaryPjax.load(Client.defaultRedirectUrl());
                 showLoadingImage(el_welcome_container, 'dark');
             }
+            not_sure.addEventListener('click', onNotSure);
 
-            const upgrade_info = Client.getUpgradeInfo();
-            const el_upgrade_title = getElementById('upgrade_title');
-            let upgrade_btn_txt = '';
+            cfd.addEventListener('click', onCFD);
 
-            if (upgrade_info.can_upgrade_to.length > 1) {
-                upgrade_btn_txt = localize('Real Account');
-            } else if (upgrade_info.can_upgrade_to.length === 1) {
-                upgrade_btn_txt = upgrade_info.type[0] === 'financial'
-                    ? localize('Financial Account')
-                    : localize('Real Account');
+            d_options.addEventListener('click', onDOptions);
+        });
+    };
+
+    const onNotSure = () => {
+        /* eslint no-unused-expressions: ["error", { "allowTernary": true }] */
+        (getCanUpgrade('iom') || (is_uk && is_unwelcome_uk))
```

Comment:
why not write a normal if condition here so we don't need to disable eslint?

Reply:
In reality I dont know why eslint doesnt permit it. anyways ill change it:)

Reply:
done